### DISCUSSION TITLE
Migrating collaborative Google Photos to Immich

### DISCUSSION THREAD

**Discussion by skjdghsdjgsdj** [immich-app/immich#14477] [Q&A]
I'd like to migrate my Google Photos albums to Immich, but every guide I've seen seems based around migrating one's own albums to Immich. My use case is more complex because I have many collaborative albums.

My Google Photos account has:

* Photos not belonging to any album, which is easy enough
* Albums I own containing *only* photos that *I* own, which `immich-go` seems capable of importing and preserving album structures
* Photos belonging to albums that I own and to which other people add their own photos (i.e., from different accounts)
* Photos belonging to albums that *other* people own and to which I and others contribute

I've tried to use `immich-go` to migrate a Google Takeout export of my own photos, but given the shared albums have photos contributed by other people, those aren't in the Takeout archive.

I only care about the photos, their date/time, and their associated albums. Comments and whatnot are irrelevant to me. Geocoding would be nice but not necessary.

What's the most reasonable and reliable way of migrating *all* of these photos to Immich while preserving the album structure, including albums created by people other than me, and with photos contributed by people other than me?

**Comment by philnagel** [comment]
If Google Takeout is not including the photos "owned" by other users, you can add them to your own account while preserving the album structure. Then re-run takeout and it should include those photos. To save them to your own Google account, open the shared album and click the "Save" button at the top-right.

![Screenshot 2025-04-30 at 09 34 49](https://github.com/user-attachments/assets/6a063789-09e0-4e60-af96-4f4b3f7dcdd3)

It's not ideal since it will use storage in your Google account and may be a lot of work if there's a lot of albums. There may a better way to do it, but this is the best I can think of.

**Reply by theofficialgman** [reply]
unfortunately this is false. the "save" button saves them to your library. so if you download your library (via takeout) they will be there but if you download the album (via takeout) they will NOT. so while you can get the photoes themselves, there seems to be no way to get the photos from other people actually in the album.

**Reply by philnagel** [reply]
You're right, I just figured that out myself. The way I ended up getting those photos was to do a Takeout and immich-go import. Then I deleted all my own photos from Google Photos. This left me with just other people's photos in those shared albums. I then did a download for each of the shared albums from the Google Photos UI and imported those into each Immich album. Fortunately I only had less than 10 albums to do this way so it wasn't too bad.
Interestingly some files I did this way were detected as duplicates by Immich, not sure how they were previously included. Maybe they were duplicated in several albums.
It also does not include all metadata of course like the Takeout does, so videos lost their datetime information for example. 
Google really does not want to make it too easy to leave.

**Comment by skjdghsdjgsdj** [comment]
I think that may work, but indeed as you pointed out it would consume a lot of space. That's particularly an issue right now given my Google account is at literally 99% full. Also I'm not sure it would retain the original author and would instead show as created by me?

I'm not sure Google has API coverage for this kind of use case, and in fact as I recall they're *dropping* a bunch of API coverage for programmatic access to users' libraries. But having said that, I presume my use case is pretty common, so seeing support for it in Immich would be great. Even some kind of rudimentary scraping method would help. This is the only thing stopping me from migrating, and I very much want to!

**Comment by freakyjoe8** [comment]
I managed to do it this way

**User owning the album in Google Photos**

- export album in Google photos via takeout (only contains photos of the user owning the album)
- import album via immich-go into immich (.\immich-go.exe upload from-google-photos --server=http://immich:2283 --api-key=**key**  takeout.zip)
- shared album to second user involved

**User the album is shared to**

- mark all pictures in shared album in Google Photos
- top right click "Save" --> now all Photos (own and foreign) are in the lib
- mark all pictures and add them to a new, temporary album. Name doesn't matter
- click on "Recently Added" filter and remove all photos added "Today" (they are copies from the owner of the album)
- now export this new temp album via Takeout too and import via immich-go with the credentials of the second user
- login to immich and you should see the original album being shared already from
- now go to the newly imported temp album, mark all pictures and add them to the shared album
- remove the temp album

At least for two people this should mirror the status of Google Photos to immich.

**Reply by mebezac** [reply]
Thanks! This works for me. One note to simplify things, in my testing, for the "User the album is shared to" you don't need to save the photos first, selecting all and adding to a new temporary album will automatically save the other user's photos to your google photos library. After that, you can remove them from the Recently Added" filter like you mention.

**Reply by ChriPo97** [reply]
Thanks guys! Your little tutorial really helped me out!

**Reply by anthonyalayo** [reply]
I tried this process, and after a lot of head scratching, **I am pretty sure it does not work.** My scenario is as follows:

1. I took a photo.
2. I partner share with my wife.
3. My wife started an album.
4. My wife added a photo that I took, which she can access via partner sharing, to the album she created.

In that scenario, when you perform this step:

> - top right click "Save" --> now all Photos (own and foreign) are in the lib

Google photos will make a COPY of a photo you already own! You can see this new copy under "recently added". If you attempt to delete it, it will tell you that it does take up space (e.g., 1.1 MB of storage).  So it is not being registered as your own photo.

And it gets even worse than that. If you do a photo search for the date of that photo (e.g., May 2, 2026), you will still only see that SINGLE photo on that date. The new photo doesn't even show up under the current day either. So it is essentially hidden from any timeline view.

In conclusion, this solution only works for naive cases where people only add photos they themselves own to shared albums. If they add photos that they obtained through partner sharing, this solution breaks.
